### ISSUE TITLE
twisted.web.wsgi doesn't set REMOTE_ADDR

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3730]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)| @dreid reported|
|-|-|
|Trac ID|trac#3730|
|Type|defect|
|Created|2009-03-31 02:01:10Z|
|Branch|https://github.com/twisted/twisted/tree/wsgi_remote_addr-3730|

REMOTE_ADDR is great, it's how you get the remote address, we should set it.

<details><summary>Searchable metadata</summary>

```
trac-id__3730 3730
type__defect defect
reporter__dreid dreid
priority__normal normal
milestone__ 
branch__branches_wsgi_remote_addr_3730 branches/wsgi_remote_addr-3730
branch_author__dreid dreid
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1238464870000000 1238464870000000
changetime__1238471313000000 1238471313000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_1"></a>|@dreid commented|
|-|-|

(In [26559]) Branching to 'wsgi_remote_addr-3730'

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_2"></a>|@dreid commented|
|-|-|

(In [26560]) Refs [#3730](https://github.com/twisted/twisted/issues/3730) - Add support for REMOTE_ADDR

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)|@dreid removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_4"></a>|@glyph set owner to @dreid|
|-|-|

Update the copyright notices (2008-2009), and when you get a clean build here, please feel free to merge:

   http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/wsgi_remote_addr-3730

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_5"></a>|@dreid commented|
|-|-|

(In [26564]) Update copyrights - Refs [#3730](https://github.com/twisted/twisted/issues/3730)

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_6"></a>|@dreid commented|
|-|-|

(In [26566]) Update copyrights - Refs [#3730](https://github.com/twisted/twisted/issues/3730)

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_7"></a>|@dreid set status to `closed`|
|-|-|

(In [26567]) Pass REMOTE_ADDR to the WSGI environ.

Fixes: [#3730](https://github.com/twisted/twisted/issues/3730)
Author: dreid
Reviewer: glyph

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
